engine/src/main/java/org/hibernate/validator/internal/metadata/location/ConstraintLocation.java

```diff
@@ -112,8 +87,33 @@ static ConstraintLocation forParameter(Executable executable, int index) {
 
 	/**
 	 * Obtains the value of this location from the parent. The type of the passed parent depends on the location type,
-	 * e.g. a bean would be passed for a {@link FieldConstraintLocation} or {@link GetterConstraintLocation} but an
+	 * e.g. a bean would be passed for a {@link PropertyConstraintLocation} but an
 	 * object array for a {@link ParameterConstraintLocation}.
 	 */
 	Object getValue(Object parent);
+
+	/**
+	 * Returns the nature of the constraint location.
+	 */
+	ConstraintLocationKind getKind();
+
+	enum ConstraintLocationKind {
```

Comment:
Wondering if we should move this one into a separate file as it's now a first class citizen.